python/CMakeLists.txt

```diff
@@ -275,6 +278,12 @@ if (PYARROW_BUNDLE_ARROW_CPP)
     ABI_VERSION ${ARROW_ABI_VERSION}
     SO_VERSION ${ARROW_SO_VERSION})
 
+  if (ARROW_GPU_FOUND)
+    bundle_arrow_lib(ARROW_GPU_SHARED_LIB
+      ABI_VERSION ${ARROW_ABI_VERSION}
+      SO_VERSION ${ARROW_SO_VERSION})
+  endif()
```

Comment:
This should fail if the library is not found rather than passing silently

Reply:
I believe fixing bundle_arrow_lib logic is not a part of this PR.